src/web-auth/captcha.js

```diff
@@ -180,42 +234,86 @@ function handleCaptchaProvider(element, options, challenge) {
     case FRIENDLY_CAPTCHA_PROVIDER:
       captchaClass = '.friendly-captcha';
       break;
+    case ARKOSE_PROVIDER:
+      captchaClass = '.arkose';
+      break;
   }
   var captchaDiv = element.querySelector(captchaClass);
 
   injectCaptchaScript(
     element,
-    { lang: options.lang, provider: challenge.provider },
-    function() {
+    {
+      lang: options.lang,
+      provider: challenge.provider,
+      clientSubdomain: challenge.clientSubdomain,
+      siteKey: challenge.siteKey
+    },
+    function (arkose) {
       var global = globalForCaptchaProvider(challenge.provider);
-      if (challenge.provider === FRIENDLY_CAPTCHA_PROVIDER) {
+      if (challenge.provider === ARKOSE_PROVIDER) {
+        var retryCount = 0;
+        arkose.setConfig({
+          onReady: function () {
+            if (arkoseConfig && arkoseConfig.onReady) {
+              arkoseConfig.onReady();
+            }
+          },
+          onCompleted: function (response) {
+            setValue(response.token);
+            if (arkoseConfig && arkoseConfig.onCompleted) {
+              arkoseConfig.onCompleted();
+            }
+          },
+          onError: function (response) {
+            if (retryCount < MAX_RETRY) {
+              setValue();
+              arkose.reset();
+              // To ensure reset is successful, we need to set a timeout here
+              setTimeout(function () {
+                arkose.run();
+              }, TIMEOUT_MS);
+              retryCount++;
+            } else {
+              setValue('BYPASS_CAPTCHA');
```

Comment:
Can we elaborate on what it means to by pass the captcha and why it's being done when the retry count exceeds the MAX_RETRY value?

Reply:
This [PR](https://github.com/auth0/auth0-server/pull/11733) would provider some more context. But the TLDR is that when Arkose fails and continues to fail after some retries we want to fail open if the tenant has set their configuration for Arkose to fail open. Sending `BYPASS_CAPTCHA` token is a small optimization in this scenario to tell auth0-server that Arkose is failing and to verify that we should fail open.

Reply:
Can we add an explanation in the code-base? I can see putting such a 'bypass' string in the open might raise questions from customers as well.